### DISCUSSION TITLE
Merging a 10x10 data table into a 1x100

### DISCUSSION THREAD

**Discussion by SMOK1N-ACES** [h2oai/datatable#3203] [Q&A] [answered]
This seems like a pretty straight forward problem but the solution is escaping me... I am trying to take a data table of 10 rows and 10 columns and merge it into a 1 column with all of the 100 values... can someone post some starter code to do this? Big Thanks!

**Comment by oleksiyskononenko** [comment, marked as answer]
Just use [dt.rbind()](https://datatable.readthedocs.io/en/latest/api/dt/rbind.html):
```python
from datatable import dt
DT_10x10 = dt.Frame([range(10)]*10)
DT_100x1 = dt.rbind(*DT_10x10, bynames=False)
```

**Comment by SMOK1N-ACES** [comment]
Thanks for the easy fix. I was struggling to implement rbind correctly.

**Reply by oleksiyskononenko** [reply]
You're welcome. Yes, the issue here is that `rbind()` tries to match the column names, if not matched, it pads columns with missings. May be we need to have an option to disable matching, i.e. simply rbind columns based on their order in the frame.

**Reply by oleksiyskononenko** [reply]
I just checked, we already have such an option. So you can simply do `DT_100x1 = dt.rbind(*DT_10x10, bynames=False)`. I've updated my original answer.

**Reply by SMOK1N-ACES** [reply]
somewhat piggybacking off of the last question... but is it possible to delete the rows that hold a certain string character or that are NA?

the pseudo code would be:

if row contains " : 
        delete row

if row == NA:
        delete row

reading the docs I can tell i need to be using an ifelse but I dont seem to be implementing it correctly. Again big thanks!

**Reply by oleksiyskononenko** [reply]
What about simply filtering some rows out?

```python
DT = dt.Frame(["a", "b", "c", "b"])
DT1 = DT[f.C0 != "b", :]
print(DT1)
```

Output
```python
   | C0   
   | str32
-- + -----
 0 | a    
 1 | c    
[2 rows x 1 column]
```

**Reply by SMOK1N-ACES** [reply]
does this work for substrings tho?

I've tried to break down where I am running into road blocks but it looks like I am just digging myself a bigger hole in regards to the bigger problem I am trying to solve.... here it is in brevity...

I am trying to take a datatable and add the column names to the individual values. From there I am trying to append every column to the first column so that the resulting datatable is only 1 column. However, in the resulting single column I don't want to include any values that originally were NA or an empty string " ".

Datatable is proving to be an incredible tool for what I am doing but I am struggling a bit with implementing the correct syntax.

**Reply by oleksiyskononenko** [reply]
Yes, same approach will work for substrings. If you need to do filtering based on a particular part of the string, just use [.slice()](https://datatable.readthedocs.io/en/latest/api/str/slice.html). If you need to regmatch something, take a look at [.match()](https://datatable.readthedocs.io/en/latest/api/re/match.html).

**Reply by SMOK1N-ACES** [reply]
```
    for col in DT.names:
        DT[col] = DT[:, ifelse(col != None, col + " / " + f[col]), pass, ifelse(col != "", col + " / " + f[col], pass)]
```
hate to keep hammering this but the syntax seems to be escaping me... the goal here is to only create a new string if the value is NOT NA or an empty string ""....

Is there some documentation to the syntax that I should try and ingest? In my head I cant help but feel like I am tackling this from a python point of view.

**Reply by oleksiyskononenko** [reply]
Just do

```python
for col in DT.names:
        DT[col] = DT[:, ifelse((f[col] == "") | (f[col] == None), f[col], col + " / " + f[col])]
```

All the documentation you need here is just [ifelse](https://datatable.readthedocs.io/en/latest/api/dt/ifelse.html).
